### PR TITLE
feat: keep auto-login session state separate from interactive login

### PR DESCRIPTION
**Pull request by patmooney**
Things are getting messy. Any suggestions welcome!

 - If login url has `?invisible=truthy` within it then the sessionKey becomes `oidc:login.vixen.com-auto`
 - if the url has `?state=<random>|auto` in it then the sessionKey becomes `oidc:login.vixen.com-auto`

### CODE REVIEW

File: test/issuer/default_http_options.test.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ describe('Issuer#defaultHttpOptions', function () {
 
   it('includes a user-agent by default', function () {
     expect(Issuer.defaultHttpOptions).to.have.nested.property('headers.User-Agent')
-      .to.match(/^openid-client/);
+      .to.match(/openid-client/);
```
Comment (andrzej-woof) [inline test/issuer/default_http_options.test.js:12]: ?

**Review by andrzej-woof — COMMENTED** [review]
I can't stop thinking that we're doing something wrong with the iframe or whatever that is the root cause of this behaviour and here we're just trying to get rid of exceptions. 
I mean it's probably fine as temporary workaround but I don't feel like it's a good long term solution.

Also as far as I can see you can pass session key to `Strategy` like:
```
    passport.use(
        'oidc',
        new Strategy({ client, params, passReqToCallback: true, sessionKey: 'something' }, securifyStratCallback)
    );
```
so wouldn't it be better to do it in frontend where we authenticate and leave iframe (that is in SSO, right?) with the default key? 🤔

File: test/issuer/default_http_options.test.js (hunk 1 above)
Comment (patmooney) [reply test/issuer/default_http_options.test.js:12]: The user-agent string is derived from the package name which was previously 'openid-client' and it is now @gmsllc/openid-client

**Comment by patmooney** [comment]
@andrzej-kodify The option I had was to have two separate passport strategies, one for the iframe and one for interactive login OR to put all the logic into our now forked+customised version of the library.

I chose to put the logic in the library to not further over-complicate the login code in frontend and in backend... I'm not sure if everybody is using the new frontend code to test stuff yet (I'm certainly not) which would mean I'd have to make the changes twice. The problems with this approach is that we are over-specifying the library for our use and adding to the black-box nature of the SSO workflow.

The fact it is auto-login also has to be communicated on the return journey in order for us to use the correct passport strategy. This can only be done by updating the "state" param which at the moment is implemented in this library and is not alterable through configuration (although i will stand corrected if otherwise true). OR we could... make the redirect URL for the iframe login go to a separate route but this route would have to be added to frontend+backend and also to the securify config (redirect URLs are pre-ordained).

Another option would be to make the `state` and `sessionKey` configurable at the start of every request so the `state` would end up looking like

<JWT>{ sessionKey: 'whatever', state: <random> }</JWT>

At the moment the library detects whether `invisible` is included in the login URL and then stashes the state in the appropriate key in the user's session. So now it is over-specified but hopefully not over-engineered.

**Review by major-mann — APPROVED** [review]

**Pull request closed without merging**